Unhide financial acls, disable if not required.

eileenmcnaughton opened the pull request:
We should also
- stop enabling on install - probably first! I can't recall where we do that...
@colemanw  I started thinking about how we would do this -

Note my other PRs working this out of core are https://github.com/civicrm/civicrm-core/pull/28967 & https://github.com/civicrm/civicrm-core/pull/28960 - 

Overview
----------------------------------------
Unhide financial acls, disable if not required.

Before
----------------------------------------
financialacls extension installed on all sites, disabled if setting not set, hidden

After
----------------------------------------
unhidden, no setting, uninstalled on sites that do not have the setting set

Technical Details
----------------------------------------
I think the main way in which core calls the extension currently is where you see isACLFinancialTypeStatus() - my most recent effort to address that is in https://github.com/civicrm/civicrm-core/pull/28967 & I guess we can't remove the setting as long as tose calls exist - hence this is still WIP - we could check if the extension is enabled - fugly but...


![image](https://github.com/civicrm/civicrm-core/assets/336308/0d9f1f8a-3292-4a7e-847e-b0764c9af23c)

Comments
----------------------------------------
If this feature were to be built now the entire thing would be outside core, community supported, using hooks & I think that is ideally where it will end up. However, it was implemented a long time ago when there was less hook support

eileenmcnaughton committed:
Unhide financial acls, disable if not required.

eileenmcnaughton commented:
@colemanw this is passing now!

colemanw commented on CRM/Upgrade/Incremental/php/FiveSeventyThree.php:
```diff
@@ -29,6 +29,24 @@ class CRM_Upgrade_Incremental_php_FiveSeventyThree extends CRM_Upgrade_Increment
+
+  public static function disableFinancialAcl($rev): bool {
+    $setting = CRM_Core_DAO::singleValueQuery('SELECT value FROM civicrm_setting WHERE name = "acl_financial_type"');
+    if (!$setting) {
+      return TRUE;
+    }
+    $setting = unserialize($setting);
+    if ($setting) {
```
@eileenmcnaughton I'm confused about what this code is doing. Why the `unserialize()`? The setting appears to be a boolean, not serialized:

https://github.com/civicrm/civicrm-core/blob/aac2d7085ebbb68dac90eca05e04b2188876bc28/ext/financialacls/tests/phpunit/Civi/Financialacls/FinancialTypeTest.php#L24

colemanw replied:
Oh, ok I guess you have to unserialize because you're pulling it directly from the database?
But now I'm still confused because the logic appears to be backward. You are uninstalling the extension if the setting is TRUE? Shouldn't it be the other way 'round?

eileenmcnaughton replied:
@colemanw yep.... it should be...

eileenmcnaughton replied:
@colemanw above is fixed

seamuslee001 commented:
note just flagging at the moment if a user wants to disable civi_contribute extension / Contribute Component the financialacls block it

eileenmcnaughton commented:
@seamuslee001 yeah - that is what this is supposed to help with - ie disable it for most users. There is another PR open to further move the logic over but struggling to get engagement on that one https://github.com/civicrm/civicrm-core/pull/28967 - possibly 'cheap & cheerful acls' should be in the core extension & only the JMA extension would have 'thorough but performance-harsh' acls but at the moment we have a bit of both in core

jitendrapurohit commented on ext/financialacls/CRM/Utils/Check/Component/FinancialTypeAcls.php:
```diff
@@ -22,9 +22,8 @@ class CRM_Utils_Check_Component_FinancialTypeAcls extends CRM_Utils_Check_Compon
-    $ftAclSetting = Civi::settings()->get('acl_financial_type');
     $financialAclExtension = civicrm_api3('extension', 'get', ['key' => 'biz.jmaconsulting.financialaclreport', 'sequential' => 1]);
-    if ($ftAclSetting && (($financialAclExtension['count'] == 1 && $financialAclExtension['values'][0]['status'] != 'installed') || $financialAclExtension['count'] !== 1)) {
+    if (($financialAclExtension['count'] == 1 && $financialAclExtension['values'][0]['status'] !== 'installed') || $financialAclExtension['count'] !== 1) {
```
@eileenmcnaughton Do we need to add a check hook in this extension https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_check/?

eileenmcnaughton replied:
so I had doubts about how valid this is now - it does depend on where we land with https://github.com/civicrm/civicrm-core/pull/28967 - ie do we 

a) do expensive, super thorough thing in in core & make the extension obsolete or
b) do the thing that performs well & is probably enough for most cases in core & recommend people use the extension for the expensive thing - I haven't  had any engagement from @JoeMurray or @seamuslee001 on that but I think the value of the check is grey enough that removing for now seems OK

eileenmcnaughton replied:
now that @seamuslee001 has merged https://github.com/civicrm/civicrm-core/pull/28967 this check can go

eileenmcnaughton commented:
This was held up because of lack of clarity around the check - but the merging of https://github.com/civicrm/civicrm-core/pull/28967 means the check / extension is no longer relevant so I have removed it. The previous review otherwise tested this so I think it should be mergeable too @seamuslee001  - I updated the upgrader version

seamuslee001 commented on CRM/Financial/BAO/FinancialType.php:
```diff
@@ -396,10 +396,23 @@ public static function checkPermissionToEditFinancialType($financialTypeID) {
    *
    * @todo rename this function e.g isFinancialTypeACLsEnabled.
    *
+   * @deprecated since 5.75 will be removed around 5.90
```
Should this have a deprecatedWarning call as well?

eileenmcnaughton replied:
We actually have core calls to it still - so the removal date is probably optimistic but we can always push it back

eileenmcnaughton commented:
Just rebased this but it was passing - should be good to merge now I think

eileenmcnaughton commented:
@colemanw can you give this MOP since all the issues are resolved now
